cmd/crinit/crinit.go

```diff
@@ -31,6 +32,9 @@ const (
 )
 
 func main() {
+	// Workaround for "ERROR: logging before flag.Parse".
+	flag.CommandLine.Parse([]string{})
```

Comment:
We can remove this based on my comment below.

Reply:
Can you explain this to me in more detail? I've made changes to the normalization function setup, but I still don't see how I can remove this line and avoid `ERROR: logging before flag.Parse`.

Reply:
Oh, sure! My comment is that I think we still want warnings, just not for the ones we already know about (glog). That is, removing this line will just avoid the warnings from the glog flags having underscores. But I think we should still be warned if we've inadvertently added a different flag with an underscore later on in the code. This is achieved because we have set the warning normalization function **after** we've normalized the known glog flags that contain underscores.

Reply:
Ah, I see. In that case, I don't think we can remove this line, since it works around a different issue. I've done the fixes necessary to get the warnings for the normalization function not to apply to the known flags, but I noticed that when running `crinit init` without this line, I still get the `ERROR: logging before flag.Parse` line. There might be another way to remove this, but I couldn't figure it out from examining the code for `kubectl`.

Reply:
Are you 100% sure you're still seeing that error? I applied this patch and removed that line and I don't see any errors.

Reply:
It looks like the reason `kubectl` doesn't run into this problem is because it's using a much older version of glog (based off the glog commit [d1c4472bf2efd3826f2b5bdcc02d8416798d678c](https://github.com/golang/glog/commit/d1c4472bf2efd3826f2b5bdcc02d8416798d678c)) that did not have [this error printed to stderr](https://github.com/golang/glog/blob/23def4e6c14b4da8ac2ed8007337bc5eb5007998/glog.go#L680).

I've seen other workarounds pull in glog based off the commit [44145f04b68cf362d9c4df2182967c2275eaefed](https://github.com/golang/glog/commit/44145f04b68cf362d9c4df2182967c2275eaefed) which is the commit right before that error was introduced by commit [65d674618f712aa808a7d0104131b9206fc3d5ad](https://github.com/golang/glog/commit/65d674618f712aa808a7d0104131b9206fc3d5ad).

So we could either do one of the above. Or if we want to keep a workaround, we can probably just call the golang `flag.Parse()` after all flags are defined and before flags are accessed by the program (per the docs) in `pkg/crinit/crinit.go` right before we return `rootCmd`.

Reply:
Unfortunately, calling flag.Parse() just before returning `rootCmd` short-circuits `cobra`'s parsing of the `-h` flag and means that the standard go flag library prints out just the flags rather than the `cobra` command help.

Sigh. I guess pinning to the same glog version as kubernetes is reasonable. I'd rather not do that in this PR with all the other work going on, so I'm going to remove the fix for that and file an issue to address it after all the vendoring stuff goes in.

Reply:
Filed #66